src/amo/searchUtils.js

```diff
@@ -66,6 +66,38 @@ export function addVersionCompatibilityToFilters({
     newFilters.compatibleWithVersion = userAgentInfo.browser.version;
   }
 
+  // This adds a filter to return only Android compatible add-ons. We cannot
+  // add this fix to `fixFiltersForClientApp()` because we need to override
+  // `newFilters.compatibleWithVersion`.
+  if (newFilters.clientApp === CLIENT_APP_ANDROID) {
+    // On Android, we only have extensions.
+    log.debug(oneLine`Setting "addonType" to "extension" for Android.`);
+    newFilters.addonType = ADDON_TYPE_EXTENSION;
+
+    if (_config.get('enableFeatureMoreAndroidExtensions')) {
+      // If the browser is not Firefox for Android, we want to filter out
+      // extensions to only the limited set, so we force-set the value of
+      // `compatibleWithVersion`. This should be removed after General
+      // Avaibility.
+      //
+      // This will apply even to Firefox Desktop browsing Android pages.
+      if (
+        userAgentInfo.browser.name !== 'Firefox' ||
+        userAgentInfo.os.name !== USER_AGENT_OS_ANDROID
+      ) {
```

Comment:
Any reason not to do (+ the relevant import at the top)

```suggestion
      if (!isFirefoxForAndroid(userAgentInfo) {
```

?

Reply:
Yeah, that breaks the build for some reason. I don't know if this is also why we're not using another similar function above.